### PR TITLE
adding S3 and KMS IAM actions to LogAndMetricsDeliveryRole to see contract test results after cfn submit

### PR DESCRIPTION
**Pull request by PatMyron**
[KMS IAM actions](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_awskeymanagementservice.html#awskeymanagementservice-actions-as-permissions)
[S3 IAM actions](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazons3.html#amazons3-actions-as-permissions)

["In order to see how the contract tests will run during registration, you can use cfn submit. You will need to add S3 and KMS permissions to your log and metrics delivery role. You can add the following policy as an inline policy to your logging role"](https://w.amazon.com/bin/view/AWS21/Design/Uluru/ContractTests/#HTestingcontracttestswithcfnsubmit)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/rpdk/core/data/managed-upload-infrastructure.yaml [hunk 1]
```diff
@@ -117,13 +117,19 @@ Resources:
...
+            - kms:GenerateDataKey*
+            - kms:DescribeKey
             - logs:CreateLogGroup
             - logs:CreateLogStream
             - logs:DescribeLogGroups
             - logs:DescribeLogStreams
             - logs:PutLogEvents
             - cloudwatch:ListMetrics
             - cloudwatch:PutMetricData
+            - s3:PutObject
```
Comment (LaikaN57) [inline src/rpdk/core/data/managed-upload-infrastructure.yaml:132]: This seems particularly dangerous to leave wide open. Can we scope it to just the bucket above?

File: src/rpdk/core/data/managed-upload-infrastructure.yaml [hunk 2]
```diff
@@ -117,13 +117,19 @@ Resources:
           Statement:
           - Effect: Allow
             Action:
+            - kms:Encrypt
+            - kms:Decrypt
```
Comment (LaikaN57) [inline src/rpdk/core/data/managed-upload-infrastructure.yaml:121]: This seems particularly dangerous to leave wide open. Can we scope it to just the KMS key above?

Comment (wbingli) [reply src/rpdk/core/data/managed-upload-infrastructure.yaml:121]: Good comment. We should have separate policy for KMS and S3, and scope down to the Key and bucket created above.

File: src/rpdk/core/data/managed-upload-infrastructure.yaml (hunk 1 above)
Comment (PatMyron) [reply src/rpdk/core/data/managed-upload-infrastructure.yaml:132]: https://github.com/aws-cloudformation/cloudformation-cli/issues/466

**Pull request closed without merging**